Sources/KlaviyoSwift/KlaviyoState.swift

```diff
@@ -88,23 +104,68 @@ struct KlaviyoState: Equatable, Codable {
         case anonymousId
         case phoneNumber
         case externalId
-        case pushToken
         case queue
+        case pushTokenData
     }
 
     mutating func enqueueRequest(request: KlaviyoAPI.KlaviyoRequest) {
-        guard queue.count + 1 < MAX_QUEUE_SIZE else {
+        guard queue.count + 1 < StateManagementConstants.maxQueueSize else {
             return
         }
         queue.append(request)
     }
 
-    mutating func enqueueProfileRequest() {
-        guard let request = try? buildProfileRequest(), let request = try? updateRequestAndStateWithPendingProfile(request: request) else {
+    mutating func updateEmail(email: String) {
+        guard email != self.email else {
             return
         }
+        self.email = email
+        enqueueProfileOrTokenRequest()
+    }
 
-        queue.append(request)
+    mutating func updateExternalId(externalId: String) {
+        guard externalId != self.externalId else {
+            return
+        }
+        self.externalId = externalId
+        enqueueProfileOrTokenRequest()
+    }
+
+    mutating func updatePhoneNumber(phoneNumber: String) {
+        guard phoneNumber != self.phoneNumber else {
+            return
+        }
+        self.phoneNumber = phoneNumber
+        enqueueProfileOrTokenRequest()
+    }
+
+    mutating func enqueueProfileOrTokenRequest() {
+        guard let apiKey = apiKey,
+              let anonymousId = anonymousId else {
+            runtimeWarn("SDK internal error")
+            return
+        }
+        // if we have push data and we are switching emails
+        // we want to associate the token with the new email.
+        if let pushTokenData = pushTokenData {
```

Comment:
```suggestion
        if let pushTokenData = self.pushTokenData {
```

NIT: so that we are consistent with using `self` when accessing this property.

Reply:
This doesn't match our linting rules 😬

Reply:
it definitely does read weird.

Reply:
with syntax highlighting it's not as bad.
![Screenshot 2023-11-29 at 4 29 43 PM](https://github.com/klaviyo/klaviyo-swift-sdk/assets/653455/66ac343f-639e-4fc2-a77f-6070ccb4b1f2)

Reply:
yeah i was thinking that